xray/conventions.py

```diff
@@ -94,7 +94,20 @@ def _unpack_netcdf_time_units(units):
 def _decode_datetime_with_netcdf4(num_dates, units, calendar):
     import netCDF4 as nc4
 
-    dates = np.asarray(nc4.num2date(num_dates, units, calendar))
+    try:
+        dates = np.asarray(nc4.num2date(num_dates, units, calendar))
+    except ValueError:
+        # num2date needs calendar year start >= 0001 C.E. (bug submitted
+        # to unidata about this)
+        since_yr_idx = units.index('since ') + 6
+        year = int(units[since_yr_idx:since_yr_idx+4])
+        assert year==0, 'this fix only works for days since year 0'
```

Comment:
I don't really like the syntax of this assertion. Better to use an if statement and raise an exception:

```python
if year == 0:
    raise ValueError('this fix only works for days since year 0')
```